app/SectionApp/TodoContainer/Actions/CreateTodoAction.php

```diff
@@ -28,6 +31,14 @@ public function __construct(
      */
     public function handle(CreateTodoValue $todo): void
     {
-        $this->createTodoTask->handle($todo);
+        try {
+            DB::beginTransaction();
+            $todoId = $this->createTodoTask->handle($todo);
+            $this->createStepTask->handle($todoId, $todo->steps);
+            DB::commit();
+        } catch (\Exception $e) {
+            DB::rollBack();
+            throw $e;
+        }
```

Comment:
Có thể viết gọn hơn ntn: https://laravel.com/docs/10.x/database#database-transactions
```suggestion
        DB::transaction(function () use ($todo) {
            $todoId = $this->createTodoTask->handle($todo);
            $this->createStepTask->handle($todoId, $todo->steps);
        });
```